implementations/rust/ockam/ockam/src/entity/profile.rs

```diff
@@ -31,40 +69,258 @@ impl Profile {
         identifier: ProfileIdentifier,
         change_history: ProfileChangeHistory,
         verification_policies: Vec<ProfileVerificationPolicy>,
+        vault: Arc<Mutex<dyn ProfileVault>>,
     ) -> Self {
         Profile {
             identifier,
             change_history,
             verification_policies,
+            vault,
         }
     }
 }
 
 impl Profile {
-    pub fn apply(&mut self, change_event: ProfileChangeEvent) {
-        if !self.verify(&change_event) {
-            return; // TODO: Throw error
+    pub fn create_with_key(
+        label: String,
+        key_type: ProfileKeyType,
+        key_purpose: ProfileKeyPurpose,
+        attributes: Option<ProfileEventAttributes>,
+        vault: Arc<Mutex<dyn ProfileVault>>,
+    ) -> Result<Self, ockam_core::Error> {
+        let attributes = attributes.unwrap_or(ProfileEventAttributes::new());
+
+        let mut v = vault.lock().unwrap();
+
+        // TODO: Should be customisable
+        let secret_attributes = SecretAttributes {
```

Comment:
Love this, we should emulate it for the caller into this.

Reply:
@mikelodder7 could you please elaborate on this?

Reply:
Similar to the above discussion. The caller just selects a set of attributes for the profile key but we hide as much key specific detail as we can.

Reply:
agree